conference/src/test/java/com/kyriba/conference/management/api/ScheduleRestControllerIntegrationTest.java

```diff
@@ -0,0 +1,300 @@
+            "  \"endTime\" : \"" + endTime.format(HHmm) + "\"\n" +
+            "}"));
+  }
+
+
+  @Test
+  public void getPresentation_ifPresentationNotExistResponseNotFound() throws Exception
+  {
+    doReturn(Optional.empty()).when(scheduleService).getPresentation(anyLong());
+
+    mockMvc.perform(get("/api/v1/schedule/presentations/13"))
+        .andExpect(status().isNotFound());
+  }
+
+
+  @Test
+  public void getPresentation_ifIncorrectIdResponseBadRequest() throws Exception
+  {
+    doThrow(mock(ConstraintViolationException.class)).when(scheduleService).getPresentation(anyLong());
+
+    mockMvc.perform(get("/api/v1/schedule/presentations/-13"))
+        .andExpect(status().isBadRequest());
+  }
+
+
+  @Test
+  public void removePresentationEndpointTest() throws Exception
+  {
+    mockMvc.perform(delete("/api/v1/schedule/presentations/1000"))
+        .andExpect(status().isNoContent())
+        .andExpect(jsonPath("$").doesNotExist());
+
+    verify(scheduleService).deletePresentation(eq(1000L));
+  }
+
+
+  @Test
+  public void removePresentation_ifIncorrectIdResponseBadRequest() throws Exception
+  {
+
+    doThrow(mock(ConstraintViolationException.class))
+        .when(scheduleService).deletePresentation(anyLong());
+
+    mockMvc.perform(delete("/api/v1/schedule/presentations/-1000"))
+        .andExpect(status().isBadRequest());
+
+    verify(scheduleService).deletePresentation(eq(-1000L));
+  }
+
+
+  @Test
+  public void getScheduleEndpointTest() throws Exception
+  {
+    TopicDto topic = new TopicDto("Concurrency", "Brian Goetz");
+    final LocalTime startTime = of(9, 0);
+    final LocalTime endTime = of(10, 0);
+    PresentationResponse presentation1 = new PresentationResponse(11, topic, startTime, endTime);
+    PresentationResponse presentation2 = new PresentationResponse(12, topic, startTime.plusHours(1), endTime.plusHours(1));
+    doReturn(asList(presentation1, presentation2)).when(scheduleService).getSchedule();
+
+    MvcResult result = mockMvc.perform(get("/api/v1/schedule"))
+        .andExpect(status().isOk())
+        .andExpect(content().contentType(APPLICATION_JSON_UTF8_VALUE))
+        .andReturn();
+
+    String contentAsString = result.getResponse().getContentAsString();
+    ScheduleController.ScheduleResponse schedule = objectMapper.readValue(contentAsString, ScheduleController.ScheduleResponse.class);
+
+    assertThat(schedule).isNotNull();
+    assertThat(schedule.getPresentations()).isNotEmpty()
+        .containsExactlyInAnyOrder(presentation1, presentation2);
+  }
+
+
+  @Test
+  public void getSchedule_empty() throws Exception
+  {
+    doReturn(emptyList()).when(scheduleService).getSchedule();
+
+    MvcResult result = mockMvc.perform(get("/api/v1/schedule"))
+        .andExpect(status().isOk())
+        .andExpect(content().contentType(APPLICATION_JSON_UTF8_VALUE))
+        .andReturn();
+
+    String contentAsString = result.getResponse().getContentAsString();
```

Comment:
Duplicated parts could be moved into dedicated method as custom assertion. In previous tests similar comment may be applied.